lib/db/task.ts

```diff
@@ -0,0 +1,114 @@
+import { start } from "repl";
+import prisma from "./prisma";
+import { Task } from "@prisma/client";
+
+/**
+ * Retrieves a single task based on provided parameters.
+ *
+ * @export
+ * @async
+ * @param {Partial<Task>} params Information about the task.
+ * @returns {Promise<Task | null>}
+ *} The retrieved task object.
+ */
+export async function getTask(params: Partial<Task>): Promise<Task | null> {
```

Comment:
You'll want to use the custom `Task` type you created in `types.ts` here, as well as in `getTasks()`

Reply:
@winstuhn

Reply:
Done! Thanks.